packages/gatsby/src/redux/machines/page-component.ts

```diff
@@ -1,17 +1,24 @@
-const {
-  Machine,
-  actions: { assign },
-} = require(`xstate`)
+import { Machine as machine, assign, AnyEventObject } from "xstate"
 
-module.exports = Machine(
+interface IContext {
+  isInBootstrap: boolean
+  componentPath: string
+  query: string
+  pages: Set<string>
+}
+
+const defaultContext: IContext = {
+  isInBootstrap: true,
+  componentPath: ``,
+  query: ``,
+  pages: new Set(``),
+}
+
+const compoenentMachine = machine<IContext>(
```

Comment:
```suggestion
const componentMachine = machine<IContext>(
```
there is small type ;)